tests/rptest/services/admin.py

```diff
@@ -553,12 +553,16 @@ def get_partitions(self,
         information like replica set assignments with core affinities.
         """
         assert (topic is None and partition is None) or \
-                (topic is not None and partition is not None)
-        assert topic or namespace is None
+                (topic is not None)
```

Comment:
it's confusing that depending on the arguments the method returns either hosted or full partitions

Reply:
i know, it is confusing, i haven't change that as this is unrelated to this PR